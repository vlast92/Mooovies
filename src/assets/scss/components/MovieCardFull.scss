@import "../variables";

.Movie-card-full {

  .poster {

    @media all and (min-width: $table-max-width + 1){
      margin-right: 2rem;
    }
    @media all and (max-width: $table-max-width) {
      width: 100%;
      flex-basis: 100%;
      text-align: center;
      margin-bottom: 2rem;
    }
  }

  .movie-info{

    h1{
      @media all and (max-width:  $table-max-width){
        text-align: center;
      }
    }
    .info-list {
      padding: 0;
      list-style: none;

      li {
        margin-bottom: .5rem;

        &:last-child {
          margin-bottom: 0;
        }

        .info-label {
          font-weight: bold;
        }

        .ratings-list {
          margin-top: .5rem;
          padding-left: 1.25rem;
        }
      }

    }
  }

  .movie-plot {
    margin-top: 2rem;

    @media all and (max-width: $table-max-width){
      text-align: justify;
      text-align-last: center;
    }
  }
}