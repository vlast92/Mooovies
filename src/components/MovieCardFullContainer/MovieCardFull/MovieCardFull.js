import React from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import baseStyles from '../../../scss/Base.module.scss';
import gridStyles from '../../../scss/Grid.module.scss';
import componentStyle from './MovieCardFull.module.scss';

import LoadIndicator from '../../LoadIndicator';

class MovieCardFull extends React.Component {
	componentDidMount() {
		this.props.onMakingGetMovieRequest(this.props.match.params.imdbID);
	}

	componentDidUpdate = prevProps => {
		const { imdbID } = this.props.match.params;

		if (imdbID !== prevProps.match.params.imdbID) {
			this.props.onMakingGetMovieRequest(imdbID);
		}
	};

	componentWillUnmount() {
		this.props.onComponentUnmount();
	}

	render() {
		const { movieInfo, movieLoading } = this.props;
		let ratings;

		if (movieInfo.Ratings) {
			ratings = (
				<ul className={componentStyle.ratingsList}>
					{movieInfo.Ratings.map(rating => (
						<li key={rating.Source}>
							{rating.Source} : {rating.Value}
						</li>
					))}
				</ul>
			);
		}

		return (
			<div className={componentStyle.component}>
				<div className={baseStyles.container}>
					{movieLoading ? (
						<LoadIndicator text="Loading movie info..." />
					) : (
						<div>
							{movieInfo.Error ? (
								<div className={baseStyles.error}>Error: {movieInfo.Error}</div>
							) : (
								<>
									<div className={gridStyles.flex}>
										{movieInfo.Poster !== 'N/A' ? (
											<div
												className={`${componentStyle.poster} ${gridStyles.size25}`}>
												<img src={movieInfo.Poster} alt={movieInfo.Title} />
											</div>
										) : (
											''
										)}
										<div
											className={`${componentStyle.info} ${gridStyles.sizeAuto}`}>
											<h1>{movieInfo.Title}</h1>
											<ul className={componentStyle.infoList}>
												<li>
													<span className={componentStyle.label}>Year:</span>{' '}
													{movieInfo.Year}
												</li>
												<li>
													<span className={componentStyle.label}>
														Released:
													</span>{' '}
													{movieInfo.Released}
												</li>
												<li>
													<span className={componentStyle.label}>Country:</span>{' '}
													{movieInfo.Country}
												</li>
												<li>
													<span className={componentStyle.label}>Genre:</span>{' '}
													{movieInfo.Genre}
												</li>
												<li>
													<span className={componentStyle.label}>Runtime:</span>{' '}
													{movieInfo.Runtime}
												</li>
												<li>
													<span className={componentStyle.label}>
														Production:
													</span>{' '}
													{movieInfo.Production}
												</li>
												<li>
													<span className={componentStyle.label}>
														Director:
													</span>{' '}
													{movieInfo.Director}
												</li>
												<li>
													<span className={componentStyle.label}>Writer:</span>{' '}
													{movieInfo.Writer}
												</li>
												<li>
													<span className={componentStyle.label}>Actors:</span>{' '}
													{movieInfo.Actors}
												</li>
												<li>
													<span className={componentStyle.label}>Awards:</span>{' '}
													{movieInfo.Awards}
												</li>
												{ratings ? (
													<li>
														<span className={componentStyle.label}>
															Ratings:
														</span>
														{ratings}
													</li>
												) : (
													''
												)}
											</ul>
										</div>
									</div>
									<div className={componentStyle.plot}>{movieInfo.Plot}</div>
								</>
							)}
						</div>
					)}
				</div>
			</div>
		);
	}
}

export default withRouter(MovieCardFull);

MovieCardFull.propTypes = {
	movieLoading: PropTypes.bool,
	movieInfo: PropTypes.object.isRequired,
	match: PropTypes.object.isRequired,
	onMakingGetMovieRequest: PropTypes.func.isRequired,
	onComponentUnmount: PropTypes.func,
};
