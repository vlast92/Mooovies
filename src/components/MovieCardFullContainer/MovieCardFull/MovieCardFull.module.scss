@import './src/scss/variables';

.component {
	.poster {
		@media all and (min-width: $table-max-width + 1) {
			margin-right: 2rem;
		}
		@media all and (max-width: $table-max-width) {
			width: 100%;
			flex-basis: 100%;
			text-align: center;
			margin-bottom: 2rem;
		}
	}

	.info {
		h1 {
			@media all and (max-width: $table-max-width) {
				text-align: center;
			}
		}
		.infoList {
			padding: 0;
			list-style: none;

			li {
				margin-bottom: 0.5rem;

				&:last-child {
					margin-bottom: 0;
				}

				.label {
					font-weight: bold;
				}

				.ratingsList {
					margin-top: 0.5rem;
					padding-left: 1.25rem;
				}
			}
		}
	}

	.plot {
		margin-top: 2rem;

		@media all and (max-width: $table-max-width) {
			text-align: justify;
			text-align-last: center;
		}
	}
}
