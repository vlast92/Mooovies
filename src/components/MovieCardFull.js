import React from "react";
import {withRouter} from "react-router-dom";
import PropTypes from 'prop-types';
import "../assets/scss/components/MovieCardFull.scss"

import LoadIndicator from "./LoadIndicator";

class MovieCardFull extends React.Component {

    componentDidMount() {

        this.props.onMakingGetMovieRequest(this.props.match.params.imdbID);
    }

    componentDidUpdate = (prevProps) => {

        const {imdbID} = this.props.match.params;

        if (imdbID !== prevProps.match.params.imdbID) {
            this.props.onMakingGetMovieRequest(imdbID);
        }
    };

    componentWillUnmount() {

        this.props.onComponentUnmount();
    }

    render() {
        const {movieInfo, movieLoading} = this.props;
        let ratings;

        if (movieInfo.Ratings) {
            ratings = <ul className="ratings-list">
                {
                    movieInfo.Ratings.map(rating =>
                        <li key={rating.Source}>{rating.Source} : {rating.Value}</li>)
                }</ul>;
        }

        return (
            <div className="Movie-card-full">
                <div className="container">
                    {
                        movieLoading ?
                            <LoadIndicator text="Loading movie info..." /> :
                            <div className="card-content">
                                {movieInfo.Error ? <div className="error">Error: {movieInfo.Error}</div> :
                                    <>
                                        <div className="card-head grid">
                                            {movieInfo.Poster !== 'N/A' ?
                                                <div className="poster size-25">
                                                    <img src={movieInfo.Poster} alt={movieInfo.Title}/>
                                                </div> : ''}
                                            <div className="movie-info size-auto">
                                                <h1>{movieInfo.Title}</h1>
                                                <ul className="info-list">
                                                    <li>
                                                        <span className="info-label">Year:</span> {movieInfo.Year}
                                                    </li>
                                                    <li>
                                                        <span className="info-label">Released:</span> {movieInfo.Released}
                                                    </li>
                                                    <li>
                                                        <span className="info-label">Country:</span> {movieInfo.Country}
                                                    </li>
                                                    <li>
                                                        <span className="info-label">Genre:</span> {movieInfo.Genre}
                                                    </li>
                                                    <li>
                                                        <span className="info-label">Runtime:</span> {movieInfo.Runtime}
                                                    </li>
                                                    <li>
                                                        <span className="info-label">Production:</span> {movieInfo.Production}
                                                    </li>
                                                    <li>
                                                        <span className="info-label">Director:</span> {movieInfo.Director}
                                                    </li>
                                                    <li>
                                                        <span className="info-label">Writer:</span> {movieInfo.Writer}
                                                    </li>
                                                    <li>
                                                        <span className="info-label">Actors:</span> {movieInfo.Actors}
                                                    </li>
                                                    <li>
                                                        <span className="info-label">Awards:</span> {movieInfo.Awards}
                                                    </li>
                                                    {ratings ?
                                                        <li>
                                                            <span className="info-label">Ratings:</span>
                                                            {ratings}
                                                        </li> : ''
                                                    }
                                                </ul>
                                            </div>
                                        </div>
                                        <div className="movie-plot">
                                            {movieInfo.Plot}
                                        </div>
                                    </>}
                            </div>
                    }
                </div>
            </div>
        );
    }
}

export default withRouter(MovieCardFull);

MovieCardFull.propTypes = {
    movieLoading: PropTypes.bool,
    movieInfo: PropTypes.object.isRequired,
    match: PropTypes.object.isRequired,
    onMakingGetMovieRequest: PropTypes.func.isRequired,
    onComponentUnmount: PropTypes.func
};