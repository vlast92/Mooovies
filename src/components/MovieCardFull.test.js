import renderer from 'react-test-renderer';
import React from "react";

import MovieCardFull from "./MovieCardFull";
import {MemoryRouter, Route, Switch} from "react-router-dom";

test("MovieCardFull renders correctly", () => {

    const dummyMovieInfo = {
        Year: "2002",
        Released: "22 Mar 2002",
        Country: "Germany, USA",
        Genre: "Action, Horror, Sci-Fi, Thriller",
        Runtime: "117 min",
        Production: "New Line Cinema",
        Director: "Guillermo del Toro",
        Writer: "Marv Wolfman (character), Gene Colan (character), David S. Goyer",
        Actors: "Wesley Snipes, Kris Kristofferson, Ron Perlman, Leonor Varela",
        Awards: "6 wins & 10 nominations.",
        Ratings: [
            {Source: "Internet Movie Database", Value: "6.7/10"},
            {Source: "Rotten Tomatoes", Value: "57%"},
            {Source: "Metacritic", Value: "52/100"}
        ],
        Poster: "https://m.media-amazon.com/images/M/MV5BOWVjZTIzNDYtNTBlNC00NTJjLTkzOTEtOTE0MjlhYzI2YTcyXkEyXkFqcGdeQXVyNTAyODkwOQ@@._V1_SX300.jpg",
        Plot: "A rare mutation has occurred within the vampire community. The Reaper. A vampire so consumed with an insatiable bloodlust that they prey on vampires as well as humans, transforming victims who are unlucky enough to survive into Reapers themselves. Now their quickly expanding population threatens the existence of vampires, and soon there won't be enough humans in the world to satisfy their bloodlust. Blade, Whistler (Yes, he's back) and an armory expert named Scud are curiously summoned by the Shadow Council. The council reluctantly admits that they are in a dire situation and they require Blade's assistance. Blade then tenuously enters into an alliance with The Bloodpack, an elite team of vampires trained in all modes of combat to defeat the Reaper threat. Blade's team and the Bloodpack are the only line of defense which can prevent the Reaper population from wiping out the vampire and human populations."
    };
    const onMakingSearchRequest = jest.fn();

    const tree = renderer
        .create(
            <MemoryRouter initialEntries={['/tt0187738']}>
                <Switch>
                    <Route path="/:imdbID">
                        <MovieCardFull
                            movieInfo={dummyMovieInfo}
                            movieLoading={false}
                            onMakingGetMovieRequest={onMakingSearchRequest}
                        />
                    </Route>
                </Switch>
            </MemoryRouter>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});