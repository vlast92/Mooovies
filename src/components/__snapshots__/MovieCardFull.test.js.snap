// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCardFull renders correctly 1`] = `
<div
  className="Movie-card-full"
>
  <div
    className="container"
  >
    <div
      className="card-content"
    >
      <div
        className="card-head grid"
      >
        <div
          className="poster size-25"
        >
          <img
            src="https://m.media-amazon.com/images/M/MV5BOWVjZTIzNDYtNTBlNC00NTJjLTkzOTEtOTE0MjlhYzI2YTcyXkEyXkFqcGdeQXVyNTAyODkwOQ@@._V1_SX300.jpg"
          />
        </div>
        <div
          className="movie-info size-auto"
        >
          <h1 />
          <ul
            className="info-list"
          >
            <li>
              <span
                className="info-label"
              >
                Year:
              </span>
               
              2002
            </li>
            <li>
              <span
                className="info-label"
              >
                Released:
              </span>
               
              22 Mar 2002
            </li>
            <li>
              <span
                className="info-label"
              >
                Country:
              </span>
               
              Germany, USA
            </li>
            <li>
              <span
                className="info-label"
              >
                Genre:
              </span>
               
              Action, Horror, Sci-Fi, Thriller
            </li>
            <li>
              <span
                className="info-label"
              >
                Runtime:
              </span>
               
              117 min
            </li>
            <li>
              <span
                className="info-label"
              >
                Production:
              </span>
               
              New Line Cinema
            </li>
            <li>
              <span
                className="info-label"
              >
                Director:
              </span>
               
              Guillermo del Toro
            </li>
            <li>
              <span
                className="info-label"
              >
                Writer:
              </span>
               
              Marv Wolfman (character), Gene Colan (character), David S. Goyer
            </li>
            <li>
              <span
                className="info-label"
              >
                Actors:
              </span>
               
              Wesley Snipes, Kris Kristofferson, Ron Perlman, Leonor Varela
            </li>
            <li>
              <span
                className="info-label"
              >
                Awards:
              </span>
               
              6 wins & 10 nominations.
            </li>
            <li>
              <span
                className="info-label"
              >
                Ratings:
              </span>
              <ul
                className="ratings-list"
              >
                <li>
                  Internet Movie Database
                   : 
                  6.7/10
                </li>
                <li>
                  Rotten Tomatoes
                   : 
                  57%
                </li>
                <li>
                  Metacritic
                   : 
                  52/100
                </li>
              </ul>
            </li>
          </ul>
        </div>
      </div>
      <div
        className="movie-plot"
      >
        A rare mutation has occurred within the vampire community. The Reaper. A vampire so consumed with an insatiable bloodlust that they prey on vampires as well as humans, transforming victims who are unlucky enough to survive into Reapers themselves. Now their quickly expanding population threatens the existence of vampires, and soon there won't be enough humans in the world to satisfy their bloodlust. Blade, Whistler (Yes, he's back) and an armory expert named Scud are curiously summoned by the Shadow Council. The council reluctantly admits that they are in a dire situation and they require Blade's assistance. Blade then tenuously enters into an alliance with The Bloodpack, an elite team of vampires trained in all modes of combat to defeat the Reaper threat. Blade's team and the Bloodpack are the only line of defense which can prevent the Reaper population from wiping out the vampire and human populations.
      </div>
    </div>
  </div>
</div>
`;
